TestRadioBrowserSearch_Real breaks CI when upstream TLS cert expires

## Problem

`TestRadioBrowserSearch_Real` in `pkg/service/bmx/radiobrowser_test.go` makes a live HTTPS request to `all.api.radio-browser.info`. When the upstream certificate expires or is otherwise invalid, the test fails and blocks the entire `make test` run:

```
=== RUN   TestRadioBrowserSearch_Real
    radiobrowser_test.go:50: RadioBrowserSearch failed: Get "https://all.api.radio-browser.info/json/stations/search?...": tls: failed to verify certificate: x509: certificate has expired or is not yet valid: current time 2026-05-25T19:13:24Z is after 2026-05-25T19:05:45Z
--- FAIL: TestRadioBrowserSearch_Real (0.33s)
```

## Root cause

The test has no guard — it always dials the live endpoint, making the build hostage to third-party infrastructure availability and certificate health.

## Fix

Add a `testing.Short()` / environment-variable guard so the test is skipped in normal `go test` and CI runs, and only executes when explicitly opted in:

```go
func TestRadioBrowserSearch_Real(t *testing.T) {
    if testing.Short() || os.Getenv("RADIOBROWSER_INTEGRATION") == "" {
        t.Skip("skipping live network test; set RADIOBROWSER_INTEGRATION=1 to run")
    }
    // ...
}
```

This is currently the only `_Real`-suffixed test in the repo. Any future tests that dial external endpoints should follow the same pattern.

--- a/pkg/service/bmx/radiobrowser_test.go
+++ b/pkg/service/bmx/radiobrowser_test.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"net/http"
 	"net/http/httptest"
+	"os"
 	"testing"
 )
 
@@ -44,6 +45,9 @@ func TestRadioBrowserSearch(t *testing.T) {
 }
 
 func TestRadioBrowserSearch_Real(t *testing.T) {
+	if testing.Short() || os.Getenv("RADIOBROWSER_INTEGRATION") == "" {
+		t.Skip("skipping live network test; set RADIOBROWSER_INTEGRATION=1 to run")
+	}
 	query := "Deutschlandfunk Kultur"
 	resp, err := RadioBrowserSearch(query)
 	if err != nil {

--- a/pkg/service/bmx/radiobrowser_test.go
+++ b/pkg/service/bmx/radiobrowser_test.go
@@ -45,7 +45,7 @@ func TestRadioBrowserSearch(t *testing.T) {
 }
 
 func TestRadioBrowserSearch_Real(t *testing.T) {
-	if testing.Short() || os.Getenv("RADIOBROWSER_INTEGRATION") == "" {
+	if os.Getenv("RADIOBROWSER_INTEGRATION") == "" {
 		t.Skip("skipping live network test; set RADIOBROWSER_INTEGRATION=1 to run")
 	}
 	query := "Deutschlandfunk Kultur"

